ResSimpy/Nexus/nexus_load_well_list.py

```
@@ -0,0 +1,100 @@
+"""Functions for loading a well list from a surface network file."""
+
+from ResSimpy.Nexus.DataModels.NexusWellList import NexusWellList
+import ResSimpy.Nexus.nexus_file_operations as nfo
+from ResSimpy.Enums.HowEnum import OperationEnum
+
+
+def load_well_lists(file_as_list: list[str], current_date: None | str,
+                    previous_well_lists: None | list | dict = None,
+                    table_start_index: int = 0) -> list[tuple[NexusWellList, int]]:
+    """Loads a well list from a surface network file.
+
+    Args:
+        file_as_list (list[str]): The surface network file as a list of strings.
+        current_date (None | str): The current date of the simulation.
+        previous_well_lists (None | list[tuple[NexusWellList, int]]):
+        The list of tuples of already loaded in NexusWellLists with their starting index. Defaults to None.
+        table_start_index (int): The starting index of the table. Defaults to 0.
+
+    Returns:
+        NexusWellList: A NexusWellList object containing the well list data.
+    """
+    if previous_well_lists is None or isinstance(previous_well_lists, dict):
```

What is the benefit of accepting a dict as an input to this function if we're just changing it to an empty list anyway?

---

It was for "safer" typing coverage. There are potential leaky paths for dicts to get through from the constraints (which are a dict but shouldn't be getting to this part of the code). Perhaps we should raise an error instead of just converting to empty list.